source/common/common/matchers.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/type/matcher/metadata.pb.h"
+#include "envoy/type/matcher/number.pb.h"
+#include "envoy/type/matcher/string.pb.h"
+
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Matchers {
+
+class DoubleMatcher {
+public:
+  DoubleMatcher(const envoy::type::matcher::DoubleMatcher& matcher) : matcher_(matcher) {}
+
+  /**
+   * Check whether the value is matched to the matcher.
+   * @param value the double value to check.
+   * @return true if it's matched otherwise false.
+   */
+  bool match(double value) const;
+
+private:
+  const envoy::type::matcher::DoubleMatcher matcher_;
+};
+
+class StringMatcher {
+public:
+  StringMatcher(const envoy::type::matcher::StringMatcher& matcher) : matcher_(matcher) {
+    if (matcher.match_pattern_case() == envoy::type::matcher::StringMatcher::kRegex) {
+      regex_ = RegexUtil::parseRegex(matcher_.regex());
+    }
+  }
+
+  /**
+   * Check whether the value is matched to the matcher.
+   * @param value the string to check.
+   * @return true if it's matched otherwise false.
+   */
+  bool match(const std::string& value) const;
+
+private:
+  const envoy::type::matcher::StringMatcher matcher_;
+  std::regex regex_;
```

Comment:
Why isn't this part of StringMatcher?

Reply:
`StringMatcher` proto has this but only as a simple string field, here I copy the parsed regex to a local variable so that it could be used directly later.